sdk/src/main/java/com/tendarts/sdk/client/TDKeysHandler.java

```diff
@@ -0,0 +1,217 @@
+package com.tendarts.sdk.client;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.tendarts.sdk.BuildConfig;
+import com.tendarts.sdk.common.Configuration;
+import com.tendarts.sdk.common.Constants;
+import com.tendarts.sdk.common.Util;
+import com.tendarts.sdk.communications.Communications;
+import com.tendarts.sdk.communications.ICommunicationObserver;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Created by luisma on 15/3/18.
+ */
+
+public class TDKeysHandler {
+
+    private static final String LOG_TAG = TDKeysHandler.class.getSimpleName();
+
+    public interface TDKeysHandlerInterface {
+        void onSuccess();
+        void onError(String reason);
+    }
+
+    public enum KeyValueKind {
+        FLOAT(0),
+        INT(1),
+        STRING(2),
+        BOOL(3),
+        LIST(4);
+
+        private int kindValue;
+
+        KeyValueKind(int kindValue) {
+            this.kindValue = kindValue;
+        }
+
+        public int getKindValue() {
+            return kindValue;
+        }
+
+    }
+
+    public static void associateKeyValueWithDevice(final Context context,
```

Comment:
For an API if we want Kotlin interoperability add @NonNull or @Nullable must be necessary.